Sources/JWTKit/RSA/RSASigner.swift

```diff
@@ -2,25 +2,34 @@ import _CryptoExtras
 import Foundation
 
 struct RSASigner: JWTAlgorithm, CryptoSigner {
-    let key: RSAKey
+    enum Key {
+        case `public`(RSA.PublicKey)
+        case `private`(RSA.PrivateKey)
+    }
```

Comment:
I would suggest storing both a public and private key in this structure instead of making it an enumeration. Yes, I know that involves making both keys optional and arguably seems like a step backwards, but the "signer" design of JWTKit (which arguably really should be split into "signer" and "verifier" types where there's only ever one relevant key type for each) makes that the more sensible option, sadly.